app/hooks/access_control_attributes.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useCallback, useEffect, useRef, useState} from 'react';
+
+import {fetchChannelAccessControlAttributes} from '@actions/remote/channel_access_control_attributes';
+import {useServerUrl} from '@context/server';
+
+// Module-level cache for access control attributes
+// The cache stores data with a timestamp to implement a TTL (time-to-live)
+const attributesCache: Record<string, {
+    data: Record<string, string[]>;
+    timestamp: number;
+}> = {};
+
+// Cache TTL in milliseconds (5 minutes)
+const CACHE_TTL = 5 * 60 * 1000;
```

Comment:
Same comment as in webapp PR - https://github.com/mattermost/mattermost/pull/30971#discussion_r2083828720

Reply:
answered in the webapp PR https://github.com/mattermost/mattermost/pull/30971#discussion_r2086439797 "Server doesn't have a TTL, it's renewed whenever cache gets invalidated. So it's fine as it is."